cip/1.accepted/CIP2017-10-17-Cypher-9.adoc

```diff
@@ -0,0 +1,126 @@
+= CIP2017-10-17 Cypher version 9
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Mats Rydberg <mats@neo4j.org>
+
+[abstract]
+.Abstract
+--
+Cypher 9, the first official standard version of Cypher as governed by the openCypher project is outlined in this proposal.
+--
+
+toc::[]
+
+== Background
+
+During the 5th openCypher Implementers Group call (oCIG 5), a proposal for versioning was presented, which included the specification of Cypher 9 as the first official standard version of Cypher.
+Cypher 9 is a draft baseline specification which references a number of sources to craft a distributed specification for Cypher.
+These sources are detailed in this document.
+
+== Proposal
+
+Cypher 9 does not have a unified single source of truth (such as a natural language specification), but is comprised of references to pieces of documentation that describe parts of the Cypher language.
+
+=== Neo4j and Standardisation Scope
+
+The most general and comprehensive such piece is the
+http://neo4j.com/docs/developer-manual/current/cypher/[Neo4j Cypher 3.2 Manual], which documents Neo4j's 3.2 version of Cypher.
+The Neo4j implementation is however not a strict subset of Cypher 9, but contains a number of features that _are not_ part of the standard.
+The link:../../docs/standardisation-scope.adoc[Standardisation Scope] document lists which parts of the Neo4j implementation that are included, and which are excluded.
+
+==== Included features
+
+Features included are specified to work as documented in the Neo4j manual.
+Additionally, user-defined functions are required to return values within the Cypher type system.
+
+The `shortestPath()` and `allShortestPaths()` special pattern syntax is kept, but we note that these constructs will be superseded by a better fitting Regular Path Query (RPQ)-inspired syntax in the future.
+
+==== Excluded features
+
+Most of the set of excluded features are old language constructs for which there are already replacements, or which are not envisioned to fit in Cypher as the language evolves.
+
+Some of the excluded features leverage syntax ideas that do not align with recent developments, but which still encode useful semantics.
+These features include the list containment predicates and `reduce()`, which leverage a function-like syntax:
+
+* `any()`
+* `all()`
+* `none()`
+* `single()`
+* `reduce()`
+
+New syntax for these constructs needs to be proposed to consider them applicable for the standard.
+
+The `exists()` function is defined in an accepted CIP, and does not need to rely on vendor documentation.
```

Comment:
Aren't we including the form `exists(n.prop)` and excluding the form `exists(pattern)`? This sentence is not very clear.

Reply:
Not as documented in the Neo4j manual, but as specified in the CIP. I thought a CIP is better than referring to the manual?

Reply:
My interpretation on the Standardisation Scope is that it only acts as an overlay of what's in Neo4j ('s manual), and doesn't reference anything else.